File: Two pointer/633. Sum of Square Numbers.py
class Solution:
    def judgeSquareSum(self, c: int) -> bool:
        left = 0
        right = int(sqrt(c))
        while left <= right:
            cur = left ** 2 + right ** 2
            if cur == c:
                return 1
            if cur < c:
                left += 1
            else:
                right -= 1
        return 0

File: Two pointer/633. Sum of Square Numbers.cpp
class Solution {
public:
    bool judgeSquareSum(int c) {
        int left = 0;
        int right = sqrt(c);
        long long cur;
        while (left <= right){
            cur = pow(left, 2) + pow(right, 2);
            if (cur == c)
                return true;
            if (cur < c)
                left++;
            else
                right--;
        }
        return false;
    }
};